lib/SILGen/ArgumentSource.cpp

```diff
@@ -217,3 +217,17 @@ void ArgumentSource::forwardInto(SILGenFunction &SGF,
   auto substLoweredType = destTL.getLoweredType().getSwiftRValueType();
   RValue(SGF, loc, substLoweredType, outputValue).forwardInto(SGF, loc, dest);
 }
+
+SILType ArgumentSource::getSILSubstRValueType(SILGenFunction &SGF) const & {
+  CanSILFunctionType funcType = SGF.F.getLoweredFunctionType();
+  CanType substType = getSubstType();
+  AbstractionPattern origType(funcType->getGenericSignature(), substType);
```

Comment:
I am not sure. So I was trying to make this the most general correct manner of performing the operation, so if I was incorrect it would just be a no-op. That is why I wanted to ask you guys. ; ).